msal-go/advanced/managed-identity.md

```diff
@@ -0,0 +1,148 @@
+
+Prior to using managed identities from MSAL GO, developers must enable them for the resources they want to use through Azure CLI or the Azure Portal.
+
+## Examples
+
+For both user-assigned and system-assigned identities, developers can use the **New** function in [managedidentity.go](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/managedidentity/managedidentity.go#L107)
+
+### System-assigned managed identities
+
+For system-assigned managed identities, simple pass **SystemAssigned()** to the **New** function
+
+```go
+mi.New(mi.SystemAssigned())
+```
+
+[AcquireToken](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/managedidentity/managedidentity.go#L216) is called with the context, resource to acquire a token for, such as `https://management.azure.com`, along with any Optionals, discussed later
+
+```go
+miClient, err := mi.New(mi.SystemAssigned())
+if err != nil {
+    log.Fatalf("failed to create a new managed identity client: %v", err)
+    return
+}
+
+accessToken, err := miClient.AcquireToken(context.Background(), "https://vault.azure.net")
+if err != nil {
+    log.Fatalf("failed to acquire token: %v", err)
+    return
+}
+```
+
+### User-Assigned Managed Identities
+
+For user-assigned managed identities, the developer needs to pass either the client ID, full resource identifier, or the object ID of the managed identity when creating [**New**](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/managedidentity/managedidentity.go#L107).
+
+Like in the case for system-assigned managed identities, [AcquireToken](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/managedidentity/managedidentity.go#L216) is called with the resource to acquire a token for, such as `https://management.azure.com`.
+
+```go
+miClient, err := mi.New(mi.UserAssignedClientID("my-client-id"))
+miClient, err := mi.New(mi.UserAssignedObjectID("my-object-id"))
+miClient, err := mi.New(mi.UserAssignedResourceID("my-resource-id"))
+
+accessToken, err := miClient.AcquireToken(context.Background(), "https://vault.azure.net")
+if err != nil {
+    log.Fatalf("failed to acquire token: %v", err)
+    return
+}
+```
+
+## Caching
+
+By default, MSAL GO supports in-memory caching  
+MSAL does not support cache extensibility for managed identity because of security concerns when using distributed cache  
+Since a token acquired for managed identity belongs to an Azure resource, using a distributed cache might expose it to the other Azure resources sharing the cache.
```

Comment:
```suggestion
Since a token acquired for a managed identity belongs to an Azure resource, using a distributed cache might expose it to other Azure resources sharing the cache.
```